sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala

```diff
@@ -14,31 +14,25 @@
  * limitations under the License.
  */
 
-package com.akkaserverless.javasdk;
+package com.akkaserverless.javasdk.impl
 
-import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions;
-import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions
+import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl
 
 import java.util.Collections;
 import java.util.Set;
 
-public interface ComponentOptions {
+trait ComponentOptions {
 
   /**
-   * @return the headers requested to be forwarded as metadata (cannot be mutated, use
-   *     withForwardHeaders)
+   * @return
+   *   the headers requested to be forwarded as metadata (cannot be mutated, use withForwardHeaders)
    */
-  Set<String> forwardHeaders();
+  def forwardHeaders(): Set[String]
```

Comment:
I would be more explicit with that `Set` type. I see that there is an import `java.util.Set` above but I'd rather have the full package here, or rename the import to JSet.